atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ReloadingCloseableContainer.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
```

Comment:
Despite it being a generic interface, it's in cassandra just because the only consumers right now are in atlasdb-cassandra (restore and cqlclient), and by the same logic of keeping the interface inner of this class, I left it in this package. 

Not strongly tied to that decision though.